### PR TITLE
London class 7/ domenico delli carri / js2 week 1

### PR DESCRIPTION
**Pull request by Domenico-cyf**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

File: mandatory/1-writers.js [hunk 1]
```diff
@@ -60,7 +60,16 @@ Exercise 1:
...
+console.log(`Hi, my name is ${writer.firstName} ${writer.lastName}. I am ${writer.age} years old, and work as a ${writer.occupation}.`);
+};
+
+// for(let i = 0; i < writers.length ; i++){
+// console.log(writers[i].firstName);
+// };
+
+// console.log("Hi, my name is ${writer.firstName} ${lastName}. I am ${writer.age} years old, and work as a ${writers.occupation}. =>  )
+ /*
 Exercise 2:
```
Comment (amellie) [inline mandatory/1-writers.js:73]: What about exercise 2 and 3?

File: mandatory/2-water-bottle.js [hunk 2]
```diff
@@ -19,22 +19,42 @@ We made a start on this here by giving you the skeleton of our object.
 You have to implement the missing features according to the specification.
 */
 
+
+
 // Here is your starting point:
 let bottle = {
```
Comment (amellie) [inline mandatory/2-water-bottle.js:25]: Have you run the test cases? A lot of them failed

File: mandatory/2-water-bottle.js [hunk 3]
```diff
@@ -19,22 +19,42 @@ We made a start on this here by giving you the skeleton of our object.
...
   },
   drink: function () {
+    if (this.volume >= 10) {
+      this.volume -= 10;
+     
+    }
     // calling this function should decrease your bottle volume by 10 units;
   },
   isFull: function () {
+    if (this.volume = 100) {
```
Comment (amellie) [inline mandatory/2-water-bottle.js]: The syntax isn't correct. ``this.volume = 100`` literally sets the volume to 100, it does not check whether the volume is 100

File: mandatory/2-water-bottle.js [hunk 4]
```diff
@@ -19,22 +19,42 @@ We made a start on this here by giving you the skeleton of our object.
...
+     
+      return true
+    }
     // this function should return true if your bottle is full;
   },
   isEmpty: function () {
+    if (this.volume = 0) {
```
Comment (amellie) [inline mandatory/2-water-bottle.js]: same as above. Can you figure out how to fix this

File: mandatory/3-groceries.js [hunk 5]
```diff
@@ -30,15 +30,35 @@ Exercise 1:
 */
 // Gather all week item names into this array
 let weeklyGroceriesToBuy = [];
+for (day in weeklyMealPlan) {
```
Comment (amellie) [inline mandatory/3-groceries.js:33]: excellent

File: mandatory/3-groceries.js [hunk 6]
```diff
@@ -30,15 +30,35 @@ Exercise 1:
...
 */
 // Gather weekend item names into this array
-let weekendGroceriesToBuy = [];
+console.log("second exercise below");
 
+let weekendGroceriesToBuy = [];
+for (day in weeklyMealPlan) {
+  weeklyMealPlan[day].forEach((ingredient) => {
+    if (
+      !weekendGroceriesToBuy.includes(ingredient) &&
```
Comment (amellie) [inline mandatory/3-groceries.js:54]: well done on checking whether the ingredient has been added or not

**Review by amellie — COMMENTED** [review]
Well done -- It seems that you didn't include mandatory exercise #4 and #9?

**Comment by Domenico-cyf** [comment]
Thank you so much Amalia! Will correct accordingly

File: mandatory/5-recipes.js [hunk 7]
```diff
@@ -24,4 +24,41 @@ You should write and log at least 5 recipes
...
+    ingredients: ["Cinnemon", "Cumin", "Cocoa"],
+  },
+];
+for (let singleRecipe of recipes) {
+  console.log(singleRecipe.title);
+  console.log(singleRecipe.servings);
+  for(let ingredient of singleRecipe.ingredients) {
+  console.log(ingredient);
+  }
+}
```
Comment (11ma) [inline mandatory/5-recipes.js:64]: This is well done Domenico, good variable names too, makes it readable and easy to understand the for loops. 🔥

**Pull request closed without merging**